pkg/importer/dataStream.go

```diff
@@ -173,10 +173,19 @@ func CopyImage(dest, endpoint, accessKey, secKey string) error {
 	return ds.copy(dest)
 }
 
-// Parse the endpoint extensions and return a slice of read-closers. The readers are in order starting with
-// the lowest level reader (eg. http) with the last entry being the highest level reader (eg, gzip).
-// Note: the .qcow2 ext is ignored however qemu files are detected based on their magic number.
-// Note: readers are not closed here.
+// Parse the endpoint extensions and return a slice of read-closers. The readers order starts with the
+// lowest level reader (eg. http). The next reader is the final reader in the chain (eg. gz). The last
```

Comment:
>  The next reader is the final reader in the chain (eg. gz).

This is true when there is 1 level of compression. For more than one, this is not true.  It might be more descriptive to say that the readers are appended in the unpacking order of operation.